ros/api/v1/hosts.py

```diff
@@ -182,6 +201,14 @@ def build_system_filters():
             filters.append(System.state.in_(modified_states))
         else:
             filters.append(System.state.in_(SYSTEM_STATES_EXCEPT_EMPTY))
+        if operating_systems := request.args.getlist('os'):
+            modified_operating_systems = []
+            for os in operating_systems:
+                os = os.upper()
```

Comment:
`os` value is already in uppercase, we can remove `upper` function and use `split` instead to get the numerical value 
```suggestion
                os = os.split(" ")[1]
```

Reply:
If we are passing `/systems?os=7.0`, then there should be no need of either of these.